### PR TITLE
Adds action features to the metastation beerkeg-nukes

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
I have seen all the bad in this world, and I have no problem seeing it again so that I might look past it and see the the good which hides beneath.

:cl: Naksu
add: Nuclear explosive-shaped beerkegs found on metastation can now be triggered with the nuclear disk and the syndicate's legacy nuclear code for a foamy surprise
/:cl:

### CODE REVIEW

**Comment by vuonojenmustaturska** [comment]
Also yes, it still contains beer. Yes, you can still fill glasses and whatnot from it.

**Comment by Kromgar** [comment]
Is this because of that one round where the HoS and captain tried to arm the beer nuke to stop the blob?

**Comment by vuonojenmustaturska** [comment]
No it's because I wanted the give the crew a reward for tricking the syndies rather than the syndies silently figuring it out when they punch the beer keg with a disk

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
I like the feature, works fine in testing, only one odd thing in the read.

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 1]
```diff
@@ -455,6 +456,50 @@
...
+	if(W.is_refillable())
+		W.afterattack(keg, user, TRUE) 	// redirect refillable containers to the keg, allowing them to be filled
+		return TRUE 										// pretend we handled the attack, too.
+	if(istype(W, /obj/item/nuke_core_container))
+		to_chat(user, "<span class='notice'>[src] has had its plutonium core removed as a part of being decommissioned.</span>")
+		return TRUE
+	return ..()
+
+/obj/machinery/nuclearbomb/beer/actually_explode()
+	var/datum/round_event_control/E = locate(/datum/round_event_control/vent_clog/beer) in SSevents.control
```
Comment (JJRcop) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm]: This should not happen if the bomb is exploded off the station Z.

File: code/modules/events/vent_clog.dm [hunk 2]
```diff
@@ -72,3 +72,26 @@
...
+	typepath = /datum/round_event/vent_clog/beer
+	max_occurrences = 0
+
+/datum/round_event/vent_clog/beer
+	reagentsAmount = 100
+
+/datum/round_event/vent_clog/beer/announce()
+	priority_announce("The scrubbers network is experiencing an unexpected surge of pressurized beer. Some ejection of contents may occur.", "Atmospherics alert")
+
+/datum/round_event/vent_clog/beer/start()
```
Comment (JJRcop) [inline code/modules/events/vent_clog.dm]: This entire override could be removed by setting randomProbability to `0` and saferChems to `list("beer")`

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 3]
```diff
@@ -455,6 +456,50 @@
...
+	if(E)
+		E.runEvent()
+	addtimer(CALLBACK(src, .proc/really_actually_explode), 100)
+
+/obj/machinery/nuclearbomb/really_actually_explode()
```
[the remark this answers is not in the document]
Comment (vuonojenmustaturska) [reply code/modules/antagonists/nukeop/equipment/nuclearbomb.dm]: oh yeah

File: code/modules/events/vent_clog.dm (hunk 2 above)
Comment (vuonojenmustaturska) [reply code/modules/events/vent_clog.dm]: the override also spawns no cockroaches

**Review by SpaceManiac — APPROVED** [review]
Works and handles offstation correctly and the real nuclear bomb still works

**Pull request merged**